test/integration/api/compact_index_test.rb

```diff
@@ -168,6 +168,23 @@ def etag(body)
     assert_equal etag(expected), @response.headers['ETag']
   end
 
+  test "/info with multiple ruby & rubygems requirements" do
+    rubygem = create(:rubygem, name: 'gemC')
+    version = create(:version, rubygem: rubygem, number: '1.0.0', info_checksum: '65ea0d',
+                               required_ruby_version: "< 2.5, >=2.2", required_rubygems_version: "~>2,< 5.7")
+    create(:dependency, :development, version: version, rubygem: @rubygem2)
+    expected = <<~END
+      ---
+      1.0.0 |checksum:b5d4045c3f466fa91fe2cc6abe79232a1a57cdf104f7a26e716e0a1e2789df78,ruby:< 2.5&>=2.2,rubygems:~>2&< 5.7
```

Comment:
> Fetching nokogiri 1.8.0 (x64-mingw32) # respectivly (x86-mingw32)
Installing nokogiri 1.8.0 (x64-mingw32)
Gem::InstallError: nokogiri requires Ruby version < 2.5, >= 2.2.
An error occurred while installing nokogiri (1.8.0), and Bundler cannot
continue.

> has everything to do with Bundler & rubygems.org having a different understand on how to serialize required_ruby_version with multiple requirements

current requirement is formatted as following, notice comma (instead of &):
> 1.6.7.rc2-x64-mingw32 mini_portile:~> 0.7.0.rc2|checksum:3b03a61d7c22883673d5bf3794019f41184c1b0b4686d99ee2e122661e99f88f,ruby:< 2.3, >= 1.9.2,rubygems:> 1.3.1

@deivid-rodriguez do you think you can check if this is still an issue? I am guessing it is.

Reply:
Basically, we need to check how does bundler picks its required ruby version.

Reply:
Without having a closer look, I think we fixed something similar on the bundler's side in https://github.com/rubygems/bundler/pull/7554.

Assuming it's the same thing, I guess if we could fix it in the rubygems.org side it'd be better because it'd be automatically picked up by all bundler versions?

Reply:
Looks the same to me. So, no issue in changing this to `&`? ie versions after 7554 support both `,` and `&`.

Reply:
Is the last sentence a question? If it is, I don't recall, I'd need to double check, but if they don't we could always revert that PR. It's not released yet.

Reply:
> Is the last sentence a question?

yes :sweat_smile: if it is not release, best to fix it here. we have used `&` for everything else. will merge this after deploying current staging. please verify if `&` will fix the issue.

Reply:
Ok, let me just friendly ping @kou in case he remembers about #7554 and can confirm this is a good path forward.

Reply:
Yeah, I had a look and it definitely looks like the same thing to me. The PR changes `bundler` behaviour to expect coma separated `ruby` and `rubygems` requirements from the API, so we'll indeed need to revert the PR.

Fixing this in rubygems.org feels definitely best! :+1:

Reply:
I think so too but this is a compact_index gem problem ( https://github.com/rubygems/compact_index/blob/426b4bba20f84fff523f7901a27b7b09eba05285/lib/compact_index/gem_version.rb#L27-L28 ) not a rubygems.org problem.
rubygems/compact_index is archived. So should we fix this in rubygems/rubygems.org?

Reply:
I was not around when all this happened, but I believe that the `compact_index` gem was merged into `rubygems.org` and now it's only used by `bundler` specs?

If this is the case, we shoud definitely do something about it, since it sounds pretty bad that bundler specs use a gem from an archived repo.

Reply:
> the compact_index gem was merged into rubygems.org

rubygems.org uses [the same library](https://github.com/rubygems/rubygems.org/blob/master/Gemfile#L42). [bundler-api](https://raw.githubusercontent.com/bundler/bundler-api/) was merged to rubygems.org. [I agree](https://github.com/rubygems/rubygems.org/pull/1669#discussion_r181571487) that we should fix it in compact_index.
@indirect do you think you can un-archive it and add to repo and the gem if no one is responsible for it.

Reply:
Oh, ok! Then we should definitely unarchive this repo and fix it there, yeah :+1:! It's an important part of the picture.

Reply:
I wonder why it was archived in the first place :thinking:.

Reply:
I have created https://github.com/rubygems/compact_index/pull/26

> I wonder why it was archived in the first place thinking.

probably because of security notifications from github